postgis/lwgeom_out_mvt.c

```diff
@@ -45,26 +45,76 @@ Datum ST_AsMVTGeom(PG_FUNCTION_ARGS)
 	elog(ERROR, "Missing libprotobuf-c");
 	PG_RETURN_NULL();
 #else
-	LWGEOM *lwgeom_in, *lwgeom_out;
+	GBOX *bounds = NULL;
+	int32_t extent = 0;
+	int32_t buffer = 0;
+	bool clip_geom = true;
 	GSERIALIZED *geom_in, *geom_out;
-	GBOX *bounds;
-	int extent, buffer;
-	bool clip_geom;
+	LWGEOM *lwgeom_in, *lwgeom_out;
+	uint8_t type = 0;
+
 	if (PG_ARGISNULL(0))
+	{
 		PG_RETURN_NULL();
-	geom_in = PG_GETARG_GSERIALIZED_P_COPY(0);
-	lwgeom_in = lwgeom_from_gserialized(geom_in);
+	}
+
 	if (PG_ARGISNULL(1))
+	{
 		elog(ERROR, "%s: parameter bounds cannot be null", __func__);
-	bounds = (GBOX *) PG_GETARG_POINTER(1);
+		PG_RETURN_NULL();
+	}
+	bounds = (GBOX *)PG_GETARG_POINTER(1);
+	if (bounds->xmax - bounds->xmin <= 0 || bounds->ymax - bounds->ymin <= 0)
+	{
+		elog(ERROR, "%s: bounds width or height cannot be 0", __func__);
```

Comment:
why? why not just return null?

Reply:
then still <=0 and =0 mixup at least needs patching :)

Reply:
I've tried but I haven't found a way to create an invalid box (where max < min). Do you prefer using  FP_LTEQ instead?

Reply:
I see that we're checking for one thing and outputting a message about failing different check. It's never a good idea.

<= 0 are usually checks for math sanity, so there are no things like div0 or negative unsigned area, so FP_* aren't that good for them. If there wasn't such a check here, your later box size shortcut will filter all the geometries, so technically we can just emit a warning here that box is too small but output a valid empty thing instead of crashing the query. WDYT?

Reply:
> so technically we can just emit a warning here that box is too small but output a valid empty thing instead of crashing the query. WDYT?

I don't like it. Passing a tiny (or invalid) bounding box is clearly an error. I'm not saying that it can't be done, but when I use ST_AsMVT functions I iterate over the geometry, not the bounding box, so in that case I'd end up with an empty tile without a way to know that this was a programming error.

> I see that we're checking for one thing and outputting a message about failing different check. It's never a good idea.

I'll change it to something along the lines of `The geometric bounds are too small`.